coffeine/transfer_learning.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from pyriemann.transfer import TLCenter, TLStretch, encode_domains
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+def _check_data(X):
+    # make proper 3d array of covariances
+    out = None
+    if X.ndim == 3:
+        out = X
+    elif X.values.dtype == 'object':
+        # first remove unnecessary dimensions,
+        # then stack to 3d data
+        values = X.values
+        if values.shape[1] == 1:
+            values = values[:, 0]
+        out = np.stack(values)
+        if out.ndim == 2:  # deal with single sample
+            assert out.shape[0] == out.shape[1]
+            out = out[np.newaxis, :, :]
+    return out
+
+
+class ReCenter(BaseEstimator, TransformerMixin):
+    """Re-center each dataset seperately for transfer learning.
+
+    The data from each dataset are re-centered to the Identity using TLCenter
+    from Pyriemann. The difference is we assume to not have target data in the
+    training sample so when the transform function is called we fit_transform
+    TLCenter on the target (test) data.
+
+    Parameters
+    ----------
+        metric : str, default='riemann'
+            The metric to compute the mean.
+    """
+    def __init__(self, domains, metric='riemann'):
+        self.domains = domains
+        self.metric = metric
+        self.re_center = TLCenter('target_domain', metric=self.metric)
```

Comment:
This should not be put in the init 

Init should just set the init parameters as attributes